packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -356,9 +358,21 @@ const ProductList = (props) => {
     /**************** Einstein ****************/
     useEffect(() => {
         if (productSearchResult) {
-            isSearch
-                ? einstein.sendViewSearch(searchQuery, productSearchResult)
-                : einstein.sendViewCategory(category, productSearchResult)
+            if (isSearch) {
+                try {
+                    einstein.sendViewSearch(searchQuery, productSearchResult)
+                } catch (err) {
+                    console.error(err)
+                }
+                activeData.sendViewSearch(searchParams, productSearchResult)
+            } else {
+                try {
+                    einstein.sendViewCategory(category, productSearchResult)
+                } catch (err) {
+                    console.error(err)
+                }
+                activeData.sendViewCategory(searchParams, category, productSearchResult)
+            }
```

Comment:
```suggestion
            if (isSearch) {
                einstein.sendViewSearch(searchQuery, productSearchResult)
                activeData.sendViewSearch(searchParams, productSearchResult)
            } else {
                einstein.sendViewCategory(category, productSearchResult)
                activeData.sendViewCategory(searchParams, category, productSearchResult)
            }
```
Einstein should catch its own errors, so we shouldn't need to wrap it.

Reply:
While testing, I encountered few times an issue where Einstein error was thrown and no active data tracking call happened, so keeping this as is.